app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -995,12 +999,20 @@ class HomeFragment : Fragment() {
 
     private fun displayWallpaperIfEnabled() {
         if (shouldEnableWallpaper()) {
-            val wallpaperManger = requireComponents.wallpaperManager
-            // We only want to update the wallpaper when it's different from the default one
-            // as the default is applied already on xml by default.
-            if (wallpaperManger.currentWallpaper != WallpaperManager.defaultWallpaper) {
-                wallpaperManger.updateWallpaper(binding.wallpaperImageView, wallpaperManger.currentWallpaper)
-            }
+            requireComponents.appStore.flow()
+                .ifChanged { state -> state.wallpaperState.currentWallpaper }
+                .onEach { state ->
+                    // We only want to update the wallpaper when it's different from the default one
+                    // as the default is applied already on xml by default.
+                    val currentWallpaper = state.wallpaperState.currentWallpaper
+                    if (currentWallpaper != WallpaperManager.defaultWallpaper) {
+                        requireComponents.wallpaperManager.updateWallpaper(
+                            binding.wallpaperImageView,
+                            currentWallpaper
+                        )
+                    }
+                }
+                .launchIn(viewLifecycleOwner.lifecycleScope)
```

Comment:
This is for my own learning. I am wondering is there any difference between doing `launchIn(viewLifecycleOwner.lifecycleScope)` vs `flowScoped(viewLifecycleOwner.lifecycleScope)`

Reply:
Just as a point of clarity, `flowScoped` accepts a `LifecycleOwner` instead of a `CoroutineScope`. `Store.flow()`, which is used here, is also used within `flowScoped`. `Store.flow()` does some automatic lifecycle tracking and will only `emit` values while the provided lifecycle is not destroyed. It's also important to note that `Store.flow` does _not_ start emitting values again if the `LifecycleOwner` moves back into ` Start`ed state.

`flowScoped` will create a new coroutine scope using the `Main` dispatcher, and then use the that new scope to launch. `launchIn` will instead just launch a coroutine using the provided scope.

Using an injected `CoroutineScope` with `launchIn` is easier to test, IMO. It allows more fine-grained control of the coroutine context because you have a direct reference to the scope, instead of on abstract usages of the test coroutine tooling like `Dispatchers.setMain` etc. I also thought it made it easier to avoid problems with the `LifecycleOwner` auto-subscription stuff mentioned at the top, by basically requiring an active `Lifecycle.lifecycleScope`.

Reply:
It's also first thing in the morning, and I thought this was in reference to the flow collection happening in `RecentSyncedTabFeature` in my other PR. Since this is the `HomeFragment`, most of the above doesn't apply as much, but I was probably following my own example from `RecentSyncedTabFeature`. Normally, I think the convention for collecting Flows in Fragments is to use `Fragment.consumeFrom`, but here I wanted a direct Flow type so I could filter it using `Flow.ifChanged`